core/src/main/java/com/linecorp/armeria/server/CorsServerErrorHandler.java

```diff
@@ -0,0 +1,47 @@
+package com.linecorp.armeria.server;
+
+import com.linecorp.armeria.common.*;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.internal.server.CorsHeaderUtil;
+import com.linecorp.armeria.server.cors.CorsService;
+
+public class CorsServerErrorHandler implements ServerErrorHandler {
+    ServerErrorHandler serverErrorHandler;
+
+    public CorsServerErrorHandler(ServerErrorHandler serverErrorHandler) {
+        this.serverErrorHandler = serverErrorHandler;
+    }
+
+    @Override
+    public @Nullable AggregatedHttpResponse renderStatus(@Nullable ServiceRequestContext ctx, ServiceConfig config, @Nullable RequestHeaders headers, HttpStatus status, @Nullable String description, @Nullable Throwable cause) {
+
+        CorsService corsService = config.service().as(CorsService.class);
```

Comment:
We could avoid calling `serverErrorHandler.renderStatus()` in more than one place by calling it in the first place. e.g.

```
AggregatedHttpResponse res =
    serverErrorHandler.renderStatus(...);

if (ctx == null) {
    return res;
}

CorsService corsService = config.service().as(CorsService.class);
if (corsService == null) {
    return res;
}

// ... inject the headers here ...

```

Reply:
Thank you for your advise.
I fixed it.